Stack Z-Slice Plugin - Version 1.0.0

Nicholas-Schaub opened the pull request:
This PR adds a plugin that takes an image collection and a filename pattern as an input, and exports an image collection where images are stacked into a volume (in tiled tif format). This plugin is needed to take images captured at different focal points and saved as separate files and combine them into a single volumetric file.

Discussions with the WIPP team suggest that they might incorporate similar functionality at the time of image collection creation, so this plugin may eventually become deprecated. However, this tool is needed in the near term to develop plugins that handle 3D data since the new image collection standard will require volumetric data to be contained in a single file.

NHotaling commented on stack-z-slices/src/main.py:
```diff
@@ -0,0 +1,129 @@
+    
+    init_time = time.time()
+    
+    ts = [t for t in files.keys()]
```
more descriptive variable name for ts would be useful

NHotaling replied:
more descriptive variable name for ts would be useful

NHotaling commented on stack-z-slices/src/main.py:
```diff
@@ -0,0 +1,129 @@
+                                                                                                                                               c,
+                                                                                                                                               t),
+                                                                                                                                               shell=True,
+                                                                                                                                               stderr=subprocess.DEVNULL))
```
So, I know this is pretty self-explanatory in terms of what its doing.... but it'd be good if you added a few more comments to describe what each for loop is for/doing.

Nicholas-Schaub commented:
Updated the filename output value to follow a new standard that includes a variable range based on input image names. Also improved multithreading and added many comments in response to @NHotaling 's comments. Also added an `r` variable for replicates.
